epictrack-web/src/components/workPlan/status/StatusContext.tsx

```diff
@@ -32,37 +48,53 @@ export const StatusProvider = ({
   const [showStatusForm, setShowStatusForm] = React.useState<boolean>(false);
   const [showApproveStatusDialog, setShowApproveStatusDialog] =
     React.useState<boolean>(false);
+  const [isCloning, setIsCloning] = React.useState<boolean>(false);
   const [status, setStatus] = React.useState<Status>();
+  const query = useSearchParams<StatusContainerRouteParams>();
+  const workId = React.useMemo(() => query.get("work_id"), [query]);
+  const { getWorkStatuses } = useContext(WorkplanContext);
+  const { groups } = useAppSelector((state) => state.user.userDetail);
 
   const onDialogClose = () => {
     setShowStatusForm(false);
+    setIsCloning(false);
+  };
+
+  const isEditable = (is_approved: boolean) => {
+    return !is_approved || groups.includes("Super User");
   };
 
   const onSave = async (data: any, callback: () => any) => {
+    const { description, posted_date, is_approved } = data;
```

Comment:
suggesting to break this into different components like using something like this:
![image](https://github.com/bcgov/EPIC.track/assets/91914654/91fef25a-04a7-48bd-953c-861076d6249f)


if statements add cognitive complexity,  it's a metric that is always nice to decrease